include/core/solid_objects_parameters.h

```diff
@@ -301,59 +304,83 @@ namespace Parameters
   RigidSolidObject<dim>::declare_parameters(ParameterHandler &prm,
                                             unsigned int      id)
   {
+    // Use ParsedFunction<dim> during parameter declaration. We need to do this
+    // to use the declare_parameters function.
+    auto translational_velocity_parsed =
+      std::make_shared<Functions::ParsedFunction<dim>>(dim);
+    auto angular_velocity_parsed =
+      std::make_shared<Functions::ParsedFunction<dim>>(3);
+
     prm.enter_subsection("solid object " + Utilities::int_to_string(id, 1));
     {
       solid_mesh.declare_parameters(prm);
 
       prm.enter_subsection("translational velocity");
-      translational_velocity.declare_parameters(prm, dim);
+      translational_velocity_parsed->declare_parameters(prm, dim);
       prm.leave_subsection();
 
       prm.enter_subsection("angular velocity");
-      angular_velocity.declare_parameters(prm, 3);
+      angular_velocity_parsed->declare_parameters(prm, 3);
       prm.leave_subsection();
 
-
-      prm.enter_subsection("center of rotation");
-      prm.declare_entry("x", "0", Patterns::Double(), "X COR");
-      prm.declare_entry("y", "0", Patterns::Double(), "Y COR");
-      prm.declare_entry("z", "0", Patterns::Double(), "Z COR");
-      prm.leave_subsection();
+      prm.declare_entry("center of rotation",
+                        "0., 0., 0.",
+                        Patterns::List(Patterns::Double()),
+                        "Solid object center of rotation");
 
       prm.declare_entry("output solid object",
                         "true",
                         Patterns::Bool(),
                         "Controls the generation of output files");
     }
     prm.leave_subsection();
+
+    // Cast to std::shared_ptr<Function<dim>> after parameter declaration
+    translational_velocity = translational_velocity_parsed;
+    angular_velocity       = angular_velocity_parsed;
```

Comment:
I don't understand why this step?

Reply:
translational_velocity and angular_velocity used to be ``Functions::ParsedFunction<dim>``. The only way to define a ParsedFunction is to parse a prm file, which is not really nice when making a test. This is why they need to be of type  ``std::shared_ptr<Function<dim>>` 

The ``Function`` class doesn't have the ``declare_parameter`` and ``parse_parameter`` functions ( they are only implemented in the ParsedFunction class) 

In essence, if I don't use those two temporary variables, I'm not able to use the ``declare_parameter`` and ``parse_parameter`` functions or I'm not able to define a random function for my test.

Reply:
Oh ok, got it

Reply:
Yeah that's a bit convoluted, but I'm not sure there's an easier solution than this other than to do a dynamic typechecking there.
I think this is ok though, Just comment it so we know why you are doing this. LIke add a comment that explains why you are declaring these shared_ptr to not use them afterwards. If Audrey asked the question, it means from the comments it was not clear.